pkg/maps/ctmap/ctmap.go

```diff
@@ -238,6 +243,76 @@ func (m *Map) DumpEntries() (string, error) {
 	return DoDumpEntries(m)
 }
 
+// Count batch dumps the Map m and returns the count of the entries.
+func (m *Map) Count() (count int, err error) {
+	global := m.mapType.isGlobal()
+	v4 := m.mapType.isIPv4()
+	switch {
+	case global && v4:
+		return countBatch[CtKey4Global](m)
+	case global && !v4:
+		return countBatch[CtKey6Global](m)
+	case !global && v4:
+		return countBatch[CtKey4](m)
+	case !global && !v4:
+		return countBatch[CtKey6](m)
+	}
+	return
+}
+
+func countBatch[T any](m *Map) (count int, err error) {
+	mx := m.MaxEntries()
+	// If we have a hash map of N = 2^n elements, then the first collision is
+	// expected [at random] when we insert around sqrt(2*N) elements. For
+	// example, for a map of size 1024, this is around 45 elements. In normal
+	// life input is not uniformly distributed, so there could be more
+	// collisions.
+	//
+	// In practice, we can expect maximum collision lengths (# of elements in a
+	// bucket ~= chunkSize) to be around 30-40. So anything like chunk_size=10%
+	// of map size should be pretty safe. If the chunkSize is not enough, then
+	// the kernel returns ENOSPC. In this case, it is possible to just set
+	// chunkSize *= 2 and try again. However, with the current chunkSize of
+	// 4096, we observe no issues dumping the maximum size of a CT map. As
+	// explained a bit below, 4096 was an optimal number considering idle
+	// memory usage and benchmarks (see commit msg).
+	//
+	// Credits to Anton for the above explanation of htab maps.
+	const chunkSize uint32 = 4096
+	chunks := int(mx / chunkSize)
+	if chunks == 0 {
+		chunks = 1
+	}
+
+	// We can reuse the following buffers as the batch lookup does not care for
+	// the contents of the map. This saves on redundant memory allocations.
+	//
+	// The following is the number of KiB total that is allocated by Go for the
+	// following buffers based on the data type:
+	//   >>> (14*4096) / 1024 # CT IPv4 map key
+	//   56.0
+	//   >>> (38*4096) / 1024 # CT IPv6 map key
+	//   152.0
+	//   >>> (56*4096) / 1024 # CT map value
+	//   224.0
+	kout := make([]T, chunkSize)
+	vout := make([]CtEntry, chunkSize)
+
+	var cursor ebpf.BatchCursor
+	for i := 0; i < chunks; i++ {
```

Comment:
Why does this still take `chunks` into account? I think just checking for `ErrKeyNotExist` is enough.